test/Microsoft.Framework.Configuration.EnvironmentVariables.Test/EnvironmentVariablesConfigurationSourceTest.cs

```diff
@@ -133,5 +133,21 @@ public void LastVariableAddedWhenMultipleEnvironmentVariablesWithSameNameButDiff
             Assert.True(!string.IsNullOrEmpty(envConfigSrc.Get("cOMMonEnv")));
             Assert.True(!string.IsNullOrEmpty(envConfigSrc.Get("CommonEnv")));
         }
+
+        [Fact]
+        public void ReplaceDoubleUnderscoreInEnvironmentVariables()
+        {
+            var dict = new Hashtable()
+                {
+                    {"data__ConnectionString", "connection"},
+                    {"SQLCONNSTR__db1", "connStr"}
+                };
+            var envConfigSrc = new EnvironmentVariablesConfigurationSource();
+
+            envConfigSrc.Load(dict);
+
+            Assert.Equal("connection", envConfigSrc.Get("data:ConnectionString"));
+            Assert.Equal("System.Data.SqlClient", envConfigSrc.Get("Data:_db1:ProviderName"));
```

Comment:
`_db1` with underscore? I think that one is a mistake as all other tests do expect `Data:db1:ProviderName`